packages/shared/src/detect.ts

```diff
@@ -19,8 +19,8 @@ export async function detectSolid(): Promise<boolean> {
     if ($hy && typeof $hy === 'object' && 'completed' in $hy && $hy.completed instanceof WeakSet)
         return true
 
-    const bodyFirstEl = document.body.firstElementChild
-    if (bodyFirstEl && bodyFirstEl.hasAttribute(DATA_HYDRATION_KEY)) return true
+    if (document.querySelectorAll('[' + DATA_HYDRATION_KEY + ']').length > 0)
+        return true
```

Comment:
Wouldn't `document.querySelector` be better here?
I only need to know if there are ANY elements like that, I don't need to search for ALL.

Reply:
sure, I can redo this 👍